MatrixSDKTests/Crypto/CryptoMachine/MXKeysQuerySchedulerUnitTests.swift

```diff
@@ -71,71 +71,75 @@ class MXKeysQuerySchedulerUnitTests: XCTestCase {
         users: Set<String>,
         completion: @escaping (Response) -> Void,
         failure: ((Swift.Error) -> Void)? = nil
-    ) {
-        Task.detached {
-            do {
-                let result = try await self.scheduler.query(users: users)
-                completion(result)
-            } catch {
-                failure?(error)
+    ) async {
+        return await withCheckedContinuation { continuation in
```

Comment:
Unrelated change but sometimes these tests fail. I think this is because calling `Task.detached` several times in a row may not result in them being executed in the same order. To solve this we make the function `async` and completed as soon as the `Task.detached` is triggered (but not when the task completes its work). Its an await on enquing rather than completing the async work.